test/es-module/test-esm-resolve-type.js

```diff
@@ -98,86 +100,148 @@ try {
-  const pkg = rel(`node_modules/${moduleName}/package.json`);
-  const esmPkg = rel(`node_modules/${moduleName}/es/package.json`);
-  const esScript = rel(`node_modules/${moduleName}/es/index.js`);
-  const cjsScript = rel(`node_modules/${moduleName}/lib/index.js`);
-
-  createDir(nmDir);
-  createDir(mDir);
-  createDir(esSubDir);
-  createDir(cjsSubDir);
-
-  const mainPkgJsonContent = {
-    type: 'commonjs',
-    main: 'lib/index.js',
-    exports: {
-      '.': {
-        'require': './lib/index.js',
-        'import': './es/index.js'
-      },
-      './package.json': './package.json',
-    }
-  };
-  const esmPkgJsonContent = {
-    type: 'module'
-  };
+    const mDir = rel(`node_modules/${moduleName}`);
+    const esSubDir = rel(`node_modules/${moduleName}/es`);
+    const cjsSubDir = rel(`node_modules/${moduleName}/lib`);
+    const pkg = rel(`node_modules/${moduleName}/package.json`);
+    const esmPkg = rel(`node_modules/${moduleName}/es/package.json`);
+    const esScript = rel(`node_modules/${moduleName}/es/index.js`);
+    const cjsScript = rel(`node_modules/${moduleName}/lib/index.js`);
 
-  fs.writeFileSync(pkg, JSON.stringify(mainPkgJsonContent));
-  fs.writeFileSync(esmPkg, JSON.stringify(esmPkgJsonContent));
-  fs.writeFileSync(esScript,
-                   'export function esm-resolve-tester() {return 42}');
-  fs.writeFileSync(cjsScript,
-                   `module.exports = { 
-                     esm-resolve-tester: () => {return 42}}`
-  );
+    createDir(nmDir);
+    createDir(mDir);
+    createDir(esSubDir);
+    createDir(cjsSubDir);
+
+    const mainPkgJsonContent = {
+      type: 'commonjs',
+      exports: {
+        '.': {
+          'require': './lib/index.js',
+          'import': './es/index.js',
+          'default': './lib/index.js'
+        },
+        './package.json': './package.json',
+      }
+    };
+    const esmPkgJsonContent = {
+      type: 'module'
+    };
+
+    fs.writeFileSync(pkg, JSON.stringify(mainPkgJsonContent));
+    fs.writeFileSync(esmPkg, JSON.stringify(esmPkgJsonContent));
+    fs.writeFileSync(
+      esScript,
+      'export function esm-resolve-tester() {return 42}'
+    );
+    fs.writeFileSync(
+      cjsScript,
+      'module.exports = {esm-resolve-tester: () => {return 42}}'
+    );
+
+    // test the resolve
+    const resolveResult = resolve(`${moduleName}`);
+    assert.strictEqual(resolveResult.format, 'module');
+    assert.ok(resolveResult.url.includes('my-dual-package/es/index.js'));
+  }
+
+  testDualPackageWithJsMainScriptAndModuleType();
+
+  // TestParameters are ModuleName, mainRequireScript, mainImportScript,
+  // mainPackageType, subdirPkgJsonType, expectedResolvedFormat, mainSuffix
+  [
+    [ 'mjs-mod-mod', 'index.js', 'index.mjs', 'module', 'module', 'module'],
+    [ 'mjs-com-com', 'idx.js', 'idx.mjs', 'commonjs', 'commonjs', 'module'],
+    [ 'mjs-mod-com', 'index.js', 'imp.mjs', 'module', 'commonjs', 'module'],
```

Comment:
Could be worth to add a test case for this also:
```suggestion
    [ 'mjs-mod-com', 'index.js', 'imp.mjs', 'module', 'commonjs', 'module'],
    [ 'cjs-mod-com', 'index.js', 'imp.cjs', 'commonjs', 'commonjs', 'commonjs'],
```